File: JavaScript/7js 1/ctat.js
/*
|--------------------------------------------------------------------------
| Check three and two
|--------------------------------------------------------------------------
|
*/

function checkThreeAndTwo(array) {
   if ([...new Set(array)].length == 2) {
      let len = array.filter(i => i == [...new Set(array)][0]).length
      return len == 2 || len == 3;
   } else {
      return false;
   }
}


function checkThreeAndTwo(array) {
   let [a, b, c, d, e] = [...array].sort();
   return a !== e && ((a === b && c === e) || (a === c && d === e));
}


function checkThreeAndTwo(array) {
   let i, counts = {
      "a": 0,
      "b": 0,
      "c": 0
   }
   for (i = 0; i < 5; i++)
      counts[array[i]]++;
   return counts["a"] ** 2 + counts["b"] ** 2 + counts["c"] ** 2 == 13;
}


function checkThreeAndTwo(array) {
   return [2, 3].includes(array.filter(val => val === array[0]).length) && new Set(array).size === 2;
}


console.log(checkThreeAndTwo(["a", "a", "a", "b", "b"]), true);
console.log(checkThreeAndTwo(["a", "c", "a", "c", "b"]), false);
console.log(checkThreeAndTwo(["a", "a", "a", "a", "a"]), false);

File: Python/7py 1/ctat.py
"""
|--------------------------------------------------------------------------
| Check three and two
|--------------------------------------------------------------------------
|
"""

def check_three_and_two(array):
   if len(set(array)) == 2:
      x = len(list(filter(lambda i: i == list(set(array))[0], array)))
      return x == 2 or x == 3
   else:
      return False


def check_three_and_two(array):
   return { array.count(x) for x in set(array) } == {2, 3}


def check_three_and_two(array):
   return sum([array.count(x) for x in array]) == 13


def check_three_and_two(array):
   return len(set(array)) == 2 and array.count(array[0]) not in [1,4]


print(check_three_and_two(["a", "a", "a", "b", "b"]), True)
print(check_three_and_two(["a", "c", "a", "c", "b"]), False)
print(check_three_and_two(["a", "a", "a", "a", "a"]), False)